lib/src/over_react_test/custom_matchers.dart

```diff
@@ -337,3 +339,203 @@ Matcher throwsPropError_Combination(String propName, String prop2Name, [String m
+    bool hasTooManyActualLogs = matchState['hasTooManyActualLogs'] ?? false;
+
+    var description = <String>[];
+
+    if (hasTooManyActualLogs) {
+      mismatchDescription.add('Expected an equal number of expected and actual '
+          'logs, but received too many actual logs');
+
+      return mismatchDescription;
+    }
+
+    // Printing the actual logs can help because if the matcher fails when passed
+    // a callback, the test prints the function rather than the return value of
+    // recordConsoleLogs
+    if (shouldPrintActualLogs) description.add('was ${actualLogs.toString()}');
+
+    if (_expectsMatch) {
+      if (_expectsSingleMatch) {
+        description.add('Expected one log match but got $matchCount. ');
+      } else {
+        description.add('Expected ${_expected.length} log matches but got $matchCount. ');
+      }
+
+      if (hasDuplicateExpectedWarnings && hasDuplicateActualWarnings) {
+        description.add('There were duplicate matches between the expected '
+            'logs and the actual logs. Ensure each expected log '
+            'is specific and unique.');
+      } else if (hasDuplicateExpectedWarnings) {
+        description.add('There were multiple actual logs matched to a single '
+            'expected warning. Ensure each expected log is specific and'
+            ' unique.');
+      } else if (hasDuplicateActualWarnings){
+        description.add('There were multiple expected logs that matched to a '
+            'single actual warning. Ensure each expected log is specific'
+            ' and unique.');
+      }
+    } else {
+        description.add('Expected no log matches but got $matchCount. ');
+    }
+
+    mismatchDescription.add(description.join(""));
+
+    return mismatchDescription;
+  }
+}
+
+/// Matcher used to check for a single `propTypes` warning.
+///
+/// [expected] can be:
+///   * `List<String>`
+///   * `List<Contains>`
+///   * `String`
+///
+/// [shouldEnforceLogCount] can be used to ensure no are not errors outside of
+/// the expected ones. Note that this will not catch warnings or logs.
+///
+/// Related: [logsPropTypeWarnings], [logsNoPropTypeWarnings]
+_LogMatcher logsPropTypeWarning(dynamic expected, {shouldEnforceLogCount = false}) => _LogMatcher.logsPropTypeWarning(expected, shouldEnforceLogCount: shouldEnforceLogCount);
+
+/// Matcher used to check for multiple `propType` warnings.
+///
+/// Related: [logsPropTypeWarning], [logsNoPropTypeWarnings]
+_LogMatcher logsPropTypeWarnings(dynamic expected, {shouldEnforceLogCount = false}) => _LogMatcher.logsPropTypeWarnings(expected, shouldEnforceLogCount: shouldEnforceLogCount);
+
+/// Matcher used enforce that there are no `propType` warnings.
+///
+/// Related: [logsPropTypeWarning], [logsPropTypeWarnings]
+_LogMatcher logsNoPropTypeWarnings({shouldEnforceLogCount = false}) => _LogMatcher.logsNoPropTypeWarnings(shouldEnforceLogCount: shouldEnforceLogCount);
```

Comment:
This is a good one for general feedback - I included that really late in the game because I ran into the issue that the matcher was opinionated in that it's alright for there to be errors outside of the ones a consumer was looking for. That made sense to me because otherwise complex stuff could be painful to test due to irrelevant noise in the logs. 

So all it does is allows consumers to specify that if there are 3 expected `propType` errors, that's all the errors that should be logged. I did add one simple test, but I should add more if people feel like it's a good addition. Do you feel like that logic is sound or nah?

Reply:
Offline Keal pointed out that this was referring to `logsNoPropTypeWarnings` (as opposed to the other functions), which certainly doesn't need this setting 😄

Reply:
After looking at implementing the change, I realized it might still be necessary to default to not enforcing actual vs expected log list length. It's definitely up for discussion, but my thoughts are that the actual logs will still capture any error that comes though, not just `propType` warnings. So if this enforces no warnings at all by default, the matcher would be more aptly named something like `logsNoErrors` rather than be a matcher specifically for `propTypes`. 

Thoughts @kealjones-wk, @greglittlefield-wf, @aaronlademann-wf, @sydneyjodon-wk?

Reply:
I think the idea of the generic LogMatcher would be a good solution to this, then propTypes can be specific to propTypes